src/base_home.h

```diff
@@ -0,0 +1,142 @@
+//Connectable -> Possible to install network module
+//NETWORKED -> Network enabled; able to form connection between command and remote system.
+
+/**
+ * Stores capacities of command system and the auxiliary systems present.
+ */
+struct command_sys
+{
+    bool network = false;  /**Is the command system wifi enabled? (cmd core must be networked)*/
+    bool auto_conn = true; /**Automaticaly connect to remote systems if able*/
+    int max_conn = 0;      /**Network capacity (in generic units).*/
+    int max_cpu;           /**Maximum processing power (in generic units.*/
+
+    std::unordered_map<furn_id, int> aux_sys;    /**Maps <auxiliary system type, # in command system>*/
+    std::unordered_map<furn_id, int> remote_sys; /**Maps <remote system type, # connected to command system*/
+};
+
+class base_home
+{
+
+    base(submap &base_map, const tripoint &coreloc);
+
+  private:
+    //########PRIVATE##########//
+    //#BASE CORE AND LOCATIONS#//
+    //#########################//
+
+    //data members
+    int base_level; /**Level of command core. Determins max # of aux sys and, use of certain features.*/
+    command_sys cmd_sys;
+
+    std::list<base_area_flag> baflag[SEEX][SEEY]; /**additional ter and fur flags specific to bases.*/
+    std::list<npc_based *> freeloaders;           /**list containing references to the NPCs staying at the base.*/
+
+    std::list<tripoint> bunks;        /**list containing locations of sleeping spots in base.*/
+    std::list<tripoint> storage_open; /**list containing locations of unclaimed/designated storage furnature.*/
+
+    //functions
+    void define_base_area(const submap &base_map, const tripoint &coreloc); //populates baflag[][]
+
+    //########PRIVATE##########//
+    //#ITEMS AND STASHED STUFF#//
+    //#########################//
+
+    //data members
+    std::unordered_map<ammotype_id, int> gun_count;  /**stores <ammo_type, #guns_that_use_them>.*/
+    std::unordered_map<ammotype_id, int> ammo_count; /**stores <ammo_types, amount>.*/
+
+    std::list<tripoint> storage_comm; /**list containing locations of communal storage.*/
+
+    //functions
+    void count_guns(); //update gun_count
+
+    //########PRIVATE##########//
+    //#RULES AND STUFF FOR NPC#//
+    //#########################//
+
+    //data members
+    int food_ration_lv; /**Severity of food rationing. 0 = no rationing. Max = 3.*/
+    int ammo_ration_lv; /**Severity of ammo rationing. 0 = no rationing. Max = 3.*/
+    int med_ration_lv;  /**Severity of medicine/first aid rationing. 0 = no rationing. Max = 3.*/
+
+    //functions
+
+  public:
+    //##########PUBLIC#########//
+    //#BASE CORE AND LOCATIONS#//
+    //#########################//
+
+    bool is_in_base(const tripoint &); //Returns true if inside defined base area.
+    void change_lv(int new_level);
+    void run_design();
+
+    inline int get_level()
```

Comment:
`inline` is redundant here. Functions defined inside a class definition are automatically `inline`.

Reply:
didn't know that. ty